packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts

```diff
@@ -0,0 +1,258 @@
+import { IComboBoxStyles, IComboBoxOptionStyles } from './ComboBox.Props';
+import {
+  ITheme,
+  IStyle,
+  mergeStyleSets,
+  FontSizes,
+  FontWeights,
+  getFocusStyle
+} from '../../Styling';
+import { memoizeFunction } from '../../Utilities';
+
+const MS_HIGHCONTRAST_ACTIVE = '@media screen and (-ms-high-contrast: active)';
+const MS_HIGHCONTRAST_BLACK_ON_WHITE = '@media screen and (-ms-high-contrast: black-on-white)';
+
+const ComboBoxHeight = '32px';
+const ComboBoxLineHeight = '30px';
+const ComboxBoxCaretDownWidth = '32px';
+const ComboBoxOptionHeight = '36px';
+
+const getDisabledStyles = (theme: ITheme): IStyle => {
+  const { semanticColors, palette } = theme;
+  return {
+    backgroundColor: semanticColors.disabledBackground,
+    borderColor: semanticColors.disabledBackground,
+    color: semanticColors.disabledText,
+    cursor: 'default',
+    [MS_HIGHCONTRAST_ACTIVE]: {
+      borderColor: 'GrayText',
+      color: 'GrayText'
+    },
+
+    [MS_HIGHCONTRAST_BLACK_ON_WHITE]: {
+      borderColor: 'GrayText',
+      color: 'GrayText'
+    },
+  };
+};
+
+const getListOptionHighContrastStyles = (theme: ITheme): IStyle => {
+  const { semanticColors, palette } = theme;
+  return {
+    [MS_HIGHCONTRAST_ACTIVE]: {
+      backgroundColor: 'Highlight',
+      borderColor: 'Highlight',
+      color: 'HighlightText',
+    },
+  };
+};
+
+export const getOptionStyles = memoizeFunction((
+  theme: ITheme,
+  customStylesForAllOptions?: Partial<IComboBoxOptionStyles>,
+  customOptionStylesForCurrentOption?: Partial<IComboBoxOptionStyles>,
+): IComboBoxOptionStyles => {
+
+  const { semanticColors, fonts, palette } = theme;
+
+  const ComboBoxoptionBackgroundSelected = semanticColors.menuItemBackgroundChecked;
+  const ComboBoxoptionBackgroundHovered = semanticColors.menuItemBackgroundHovered;
+
+  const optionStyles: IComboBoxOptionStyles = {
+    option: [
+      {
+        background: 'transparent',
+        boxSizing: 'border-box',
+        cursor: 'pointer',
+        display: 'block',
+        width: '100%',
+        height: 'auto',
+        minHeight: ComboBoxOptionHeight,
+        lineHeight: '20px',
+        padding: '5px 16px',
+        position: 'relative',
+        border: '1px solid transparent',
+        wordWrap: 'break-word',
+        overflowWrap: 'break-word',
+        textAlign: 'left',
+
+        [MS_HIGHCONTRAST_ACTIVE]: {
+          borderColor: 'Background'
+        }
+      },
+      getFocusStyle(theme),
+    ],
+    optionHovered: {
+      backgroundColor: ComboBoxoptionBackgroundHovered,
+      color: palette.black,
+      ...getListOptionHighContrastStyles(theme)
+    },
+    optionFocused: {
+      backgroundColor: ComboBoxoptionBackgroundHovered
+    },
+    optionActive: {
+      backgroundColor: ComboBoxoptionBackgroundHovered,
+      color: palette.black

```

Comment:
please also put these colors at the top, even if they aren't using semantic slots